### PR TITLE
Support hostname, use port number other than 102, disabling ping chec…

### PR DESCRIPTION
**Pull request by sam0737**
PR for #77

### CODE REVIEW

File: S7.Net/PLC.cs [hunk 1]
```diff
@@ -147,8 +176,7 @@ public ErrorCode Open()
                 _mSocket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
                 _mSocket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.ReceiveTimeout, 1000);
                 _mSocket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.SendTimeout, 1000);
-                IPEndPoint server = new IPEndPoint(IPAddress.Parse(IP), 102);
```
Comment (mesta1) [inline S7.Net/PLC.cs]: This can not be removed, as it doesn't compile on .UWP.

File: S7.Net/PLC.cs [hunk 2]
```diff
@@ -98,25 +108,44 @@ public bool IsConnected
...
+        /// You need slot > 0 if you are connecting to external ethernet card (CP).
+        /// For S7-300 and S7-400 the default is rack = 0 and slot = 2.
+        /// </summary>
+        /// <param name="cpu">CpuType of the plc (select from the enum)</param>
+        /// <param name="host">Ip address/host of the plc</param>
+        /// <param name="port">port of the plc</param>
+        /// <param name="rack">rack of the plc, usually it's 0, but check in the hardware configuration of Step7 or TIA portal</param>
+        /// <param name="slot">slot of the CPU of the plc, usually it's 2 for S7300-S7400, 0 for S7-1200 and S7-1500.
+        /// <param name="pingTest">Do ping test before connecting the PLC</param>
+        public Plc(CpuType cpu, string host, int port, Int16 rack, Int16 slot, bool pingTest)
```
Comment (sam0737) [inline S7.Net/PLC.cs]: I just want keep the API behavior hence this design.

File: S7.Net/PLC.cs (hunk 1 above)
Comment (sam0737) [inline S7.Net/PLC.cs]: Oops I didn't know this isn't support on UWP.

**Comment by mesta1** [comment]
Ping has been merged as part of https://github.com/killnine/s7netplus/issues/87

**Pull request closed without merging**